### PR TITLE
HW8_Hanna_Pidchasiuk

### PR DESCRIPTION
**Pull request by pidchasiuk**
1. Implement your own print function. It should work on all operating systems.
2. Implement your realization of the function filter.
3. Implement your own implementation of the function map.
4. Implement your own implementation of function max and min (* additional argument amount of result, if you pass 2, function should return 2 max values from the list).
5. Implement your own all function.

### CODE REVIEW

File: HannaPidchasiuk/HW8/task1.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+            temp_line += str(item)
+        else:
+            temp_line += str(item) + " "
+    sys.stdout.write(temp_line)
+    sys.stdout.write("\n")
+
+
+if __name__ == "__main__":
+    my_print(1, "sdfg\n", (3, 4), True, 2.333, [1, 2])
+    print(1, "sdfg\n", (3, 4), True, 2.333, [1, 2])
```
Comment (bohdandovbsyh) [inline HannaPidchasiuk/HW8/task1.py:18]: good solution. I like it

File: HannaPidchasiuk/HW8/task4.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Implement your own implementation of function max and min
+# (* additional argument amount of result, if you pass 2, function should return 2 max values from the list)
+
+
+def my_max(arr: list, counter=1) -> list:
```
Comment (bohdandovbsyh) [inline HannaPidchasiuk/HW8/task4.py]: counter: int = 1

File: HannaPidchasiuk/HW8/task4.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                temp_max = arr[index]
+            else:
+                continue
+        result.append(temp_max)
+        arr.remove(temp_max)
+        counter -= 1
+    return result
+
+
+def my_min(arr: list, counter=1) -> list:
```
Comment (bohdandovbsyh) [inline HannaPidchasiuk/HW8/task4.py]: counter: int = 1

File: HannaPidchasiuk/HW8/task3.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Implement your own implementation of the function map
+
+
+def my_map(callback, sequence) -> list:
```
Comment (bohdandovbsyh) [inline HannaPidchasiuk/HW8/task3.py]: please add anotation

File: HannaPidchasiuk/HW8/task2.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# Implement your realization of the function filter
+def my_filter(callback, sequence) -> list:
```
Comment (bohdandovbsyh) [inline HannaPidchasiuk/HW8/task2.py]: please add anotations

File: HannaPidchasiuk/HW8/task2.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    return [item for item in sequence if callback(item)]
+
+
+def check_is_true(item):
```
Comment (bohdandovbsyh) [inline HannaPidchasiuk/HW8/task2.py]: add anotations

Comment (bohdandovbsyh) [reply HannaPidchasiuk/HW8/task2.py]: or even better implement lambda function to pactice

Comment (pidchasiuk) [reply HannaPidchasiuk/HW8/task2.py]: fixed

File: HannaPidchasiuk/HW8/task2.py (hunk 5 above)
Comment (pidchasiuk) [reply HannaPidchasiuk/HW8/task2.py]: fixed

File: HannaPidchasiuk/HW8/task3.py (hunk 4 above)
Comment (pidchasiuk) [reply HannaPidchasiuk/HW8/task3.py]: fixed

File: HannaPidchasiuk/HW8/task4.py (hunk 3 above)
Comment (pidchasiuk) [reply HannaPidchasiuk/HW8/task4.py]: fixed

File: HannaPidchasiuk/HW8/task4.py (hunk 2 above)
Comment (pidchasiuk) [reply HannaPidchasiuk/HW8/task4.py]: fixed

**Review by bohdandovbsyh — APPROVED** [review]

**Pull request merged**